racket/collects/racket/port.rkt

```diff
@@ -164,6 +166,66 @@
 (define (call-with-input-bytes str proc)
   (with-input-from-x 'call-with-input-bytes 1 #t str proc))
 
+(define (combine-output-ports what . with)
+  (define port-list (cons what with))
+  (map (lambda (p)
+         (or (output-port? p)
+             (raise-argument-error 'combine-output-ports "output-port?" p))) port-list)
+  (define wrap-evt-with-evts
+    (lambda (evt . evts)
+      (if (equal? (length evts) 0)
+          evt
+          (wrap-evt evt (let ([wrapped-evts (apply wrap-evt-with-evts evts)])
+                          (lambda (v)
+                            (sync wrapped-evts)))))))
+  (make-output-port
+   'combined-output-port
+   ; evt
+   (apply wrap-evt-with-evts port-list)
+   ; write-out 
+   (lambda (s start end non-block? breakable?)
+     (let ([s (subbytes s start end)])
+       (if non-block?
+           (map (lambda (p)
+                  (write-bytes-avail* s p)) port-list)
+           (begin
+             (if breakable?
+                 (map (lambda (p)
+                        (parameterize-break #t (write-bytes s p))) port-list)
+                 (map (lambda (p)
+                        (write-bytes s p)) port-list))
+             (bytes-length s)))))
+   ; close
+   void
+   ; write-out-special
+   (and (andmap port-writes-special? port-list)
+        (lambda (v no-buffer&block? breakable?)
+          (if no-buffer&block?
```

Comment:
I think this might be clearer by moving the conditional inside the `map` and using `cond`.